common/test/src/ui/tst_MapDocument.cpp

```diff
@@ -462,6 +462,44 @@ TEST_CASE_METHOD(MapDocumentTest, "canUpdateLinkedGroups")
     document->canUpdateLinkedGroups(kdl::vec_static_cast<mdl::Node*>(entityNodes)));
 }
 
+TEST_CASE_METHOD(MapDocumentTest, "newDocument")
+{
+  SECTION("Default worldspawn properties")
+  {
+    // create "new" document
+    document->newDocument(mdl::MapFormat::Standard, document->worldBounds(), game)
+      | kdl::transform_error([](auto e) { throw std::runtime_error{e.msg}; });
+
+    // workaround: because newDocument() load entities defs from a file we need to load
+    // them manually, also we can't use newDocument() because it wipes out everything
+    auto definitionWithDefaultsOwner = std::make_unique<mdl::BrushEntityDefinition>(
+      mdl::EntityPropertyValues::WorldspawnClassname,
+      Color{},
+      "",
+      std::vector<std::shared_ptr<mdl::PropertyDefinition>>{
+        std::make_shared<mdl::StringPropertyDefinition>(
+          "some_default_prop", "", "", !true(readOnly), "value"),
+      });
+    document->setEntityDefinitions(kdl::vec_from<std::unique_ptr<mdl::EntityDefinition>>(
+      std::move(definitionWithDefaultsOwner)));
+
+    // set default properties
+    document->world()->entityPropertyConfig().setDefaultProperties = true;
+    document->setWorldDefaultProps();
```

Comment:
You can't solve it by calling this function manually. The test claims to test the `newDocument` function (like it should), but you are really testing this function, which should not be public in the first place.

You could try not using `MapDocumentTest` and using the `newMapDocument` helper function from `TestUtils.h`.